nqeit-cross-engine/src/test/java/org/projectnessie/integtests/crossengine/ITCrossEngineExternalNessie.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.integtests.crossengine;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.spark.sql.SparkSession;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.projectnessie.integtests.dremio.DremioHelper;
+import org.projectnessie.integtests.dremio.IcebergDremioExtension;
+import org.projectnessie.integtests.flink.Flink;
+import org.projectnessie.integtests.flink.FlinkHelper;
+import org.projectnessie.integtests.flink.IcebergFlinkExtension;
+import org.projectnessie.integtests.iceberg.spark.IcebergSparkExtension;
+import org.projectnessie.integtests.iceberg.spark.Spark;
+import org.projectnessie.integtests.nessie.NessieTestsExtension;
+
+@ExtendWith({
+  IcebergSparkExtension.class,
+  IcebergFlinkExtension.class,
+  NessieTestsExtension.class,
+  IcebergDremioExtension.class
+})
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class ITCrossEngineExternalNessie {
+
+  private static SparkSession spark;
```

Comment:
If we would have not used static fields then all the three parameters needs to be injected in every method along with other arguments . So to avoid that headache and the clarity of code, we have used the static fields instead of injecting them multiple times. 

for eg: In `assertSelectFromEngines()` method, we need to pass along 3 parameters all the time